axelrod/strategies/meta.py

```diff
@@ -533,3 +535,80 @@ def __init__(self):
                 <= 1]
         super().__init__(team=team)
         self.classifier["long_run_time"] = False
+
+class MemoryDecay(MetaPlayer):
+    """
+    A player utilizes the (default) Tit for Tat strategy for the first (default) 15 turns,
+    at the same time memorizing the opponent's decisions. After the 15 turns have
+    passed, the player calculates a 'net cooperation score' (NCS) for his opponent,
+    weighing decisions to Cooperate as (default) 1, and to Defect as (default)
+    -2. If the opponent's NCS is below 0, the play defects; otherwise,
+    they cooperate.
+
+    The player's memories of the opponent's decisions have a random chance to be
+    altered (i.e., a C decision becomes D or vice versa; default probability
+    is 0.03) or deleted (default probability is 0.1).
+
+    It's necessary to specify EXACT name of the starting strategy if changing
+    the default. Possible strategies can be accessed with the .team attribute.
+
+    Name: Memory Decay
+    """
+    name = 'Memory Decay'
+    classifier = {
+        'memory_depth' : float('inf'),
+        'long_run_time' : False,
+        'stochastic' : True,
+        'makes_use_of' : set([]),
+        'inspects_source' : False,
+        'manipulates_source' : False,
+        'manipulates_state' : False
+    }
+
+    def __init__(self, p_memory_delete: float = 0.1, p_memory_alter: float = 0.03,
+                 loss_value: float = -2, gain_value: float = 1,
+                 memory: list = None, start_strategy: Player = TitForTat,
+                 start_strategy_duration: int = 15):
+        super().__init__()
+        self.p_memory_delete = p_memory_delete
+        self.p_memory_alter = p_memory_alter
+        self.loss_value = loss_value
+        self.gain_value = gain_value
+        self.memory = [] if memory == None else memory
+        self.start_strategy_duration = start_strategy_duration
+        self.team = start_strategy()
```

Comment:
There are numerous unit tests that are failing. This is because `self.team` needs to be a list (even if it only contains 1 strategy):

```
self.team = [start_strategy()]
```

There's information on running tests here: http://axelrod.readthedocs.io/en/stable/tutorials/contributing/running_tests.html#running-tests

but let me know if you'd like a hand :)

Reply:
I've changed `self.team` to a list and ran `python -m unittest discover`.

There are a few errors I don't  know how to handle, so I'd appreciate it if you'd take a look.
 
```
1 > python -m unittest axelrod.tests.strategies.test_meta
.F.F..E..........................................................................................................................................................
======================================================================
ERROR: test_strategy (axelrod.tests.strategies.test_meta.TestMemoryDecay)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/denis/Documents/Axelrod/axelrod/tests/strategies/test_meta.py", line 633, in test_strategy
    init_kwargs = {'start_strategy': 'Defector'})
  File "/home/denis/Documents/Axelrod/axelrod/tests/strategies/test_player.py", line 490, in versus_test
    player = self.player(**init_kwargs)
  File "/home/denis/Documents/Axelrod/axelrod/strategies/meta.py", line 579, in __init__
    self.team = [start_strategy()]
TypeError: 'str' object is not callable

======================================================================
FAIL: test_initialisation (axelrod.tests.strategies.test_meta.TestMemoryDecay)
Test that the player initiates correctly.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/denis/Documents/Axelrod/axelrod/tests/strategies/test_player.py", line 372, in test_initialisation
    self.classifier_test(self.expected_class_classifier)
  File "/home/denis/Documents/Axelrod/axelrod/tests/strategies/test_player.py", line 512, in classifier_test
    self.assertEqual(expected_class_classifier, self.player.classifier)
AssertionError: {'mem[72 chars]of': {'length', 'game'}, 'inspects_source': Fa[56 chars]alse} != {'mem[72 chars]of': set(), 'inspects_source': False, 'manipul[43 chars]alse}
  {'inspects_source': False,
   'long_run_time': False,
-  'makes_use_of': {'length', 'game'},
+  'makes_use_of': set(),
   'manipulates_source': False,
   'manipulates_state': False,
   'memory_depth': inf,
   'stochastic': True}

======================================================================
FAIL: test_repr (axelrod.tests.strategies.test_meta.TestMemoryDecay)
Test that the representation is correct.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/denis/Documents/Axelrod/axelrod/tests/strategies/test_player.py", line 377, in test_repr
    self.assertEqual(str(self.player()), self.name)
AssertionError: 'Memory Decay: 1 player' != 'MemoryDecay'
- Memory Decay: 1 player
+ MemoryDecay


----------------------------------------------------------------------
Ran 161 tests in 56.801s

FAILED (failures=2, errors=1)
```

Also, thank you for being so patient and helpful. Really means a lot to me, being new to all this :)

Reply:
> Also, thank you for being so patient and helpful. Really means a lot to me, being new to all this :)

No problem at all, appreciate the contribution. We're not far now, hopefully below I've pointed out everything that needs fixing. Not a problem if it's still not quite there though and we need to figure something else out :) :+1: